### PR TITLE
New strategy StrategyForceLatestVersions

### PR DESCRIPTION
**Pull request by Michaelpalacce**
### Description

This strategy will force you to upload the same or newer version of a package, otherwise it will fail the build, allowing us for 
better CI/CD pipelines, where we can ensure that the latest versions are always used on the server.

### Checklist

- [x] I have added relevant error handling and logging messages to help troubleshooting
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation, relevant usage information (if applicable)
- [x] I have updated CHANGELOG.md with a short summary of the changes introduced
- [x] I have tested against live environment, if applicable
- [x] I have synced any structure and/or content vRA-NG improvements with vra-ng and ts-vra-ng archetypes (if applicable)
- [x] I have my changes rebased and squashed to the minimal number of relevant commits. **Notice: don't squash all commits**
- [x] I have added a descriptive commit message with a short title, including a `Fixed #XXX -` or `Closed #XXX -` prefix to auto-close the issue

### Testing

Here are the outputs:

```log
[INFO] STRATEGY| INFO | Apply Configuration strategies for import
[INFO] STRATEGY| PASS | Source.Version > Destination.Version
[INFO] PACKAGE | SKIP | com.vmware.pscoe.library.ts.logging (2.1.0) > (2.1.0)
[INFO] PACKAGE | PASS | com.vmware.pscoe.library.ecmascript (2.38.2-SNAPSHOT) > (2.38.2-SNAPSHOT)
[INFO] PACKAGE | FAIL | com.test.test (1.0.3-SNAPSHOT) > (1.0.5-SNAPSHOT)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  34.178 s
[INFO] Finished at: 2024-05-16T18:18:24+03:00
```

```log
[INFO] STRATEGY| INFO | Apply Configuration strategies for import
[INFO] STRATEGY| PASS | Source.Version > Destination.Version
[INFO] PACKAGE | SKIP | com.vmware.pscoe.library.ts.logging (2.1.0) > (2.1.0)
[INFO] PACKAGE | PASS | com.vmware.pscoe.library.ecmascript (2.38.2-SNAPSHOT) > (2.38.2-SNAPSHOT)
[INFO] PACKAGE | PASS | com.test.test (1.0.5-SNAPSHOT) > (1.0.4-SNAPSHOT)
[INFO] Package | IMPORT | com.vmware.pscoe.library.ecmascript-2.38.2-SNAPSHOT.package
[INFO] Package | IMPORT | com.test.test-1.0.5-SNAPSHOT.package
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
```

```log
[INFO] STRATEGY| INFO | Apply Configuration strategies for import
[INFO] STRATEGY| PASS | Source.Version > Destination.Version
[INFO] PACKAGE | SKIP | com.vmware.pscoe.library.ts.logging (2.1.0) > (2.1.0)
[INFO] PACKAGE | PASS | com.vmware.pscoe.library.ecmascript (2.38.2-SNAPSHOT) > (2.38.2-SNAPSHOT)
[INFO] PACKAGE | PASS | com.test.test (1.0.4-SNAPSHOT) > (1.0.4-SNAPSHOT)
[INFO] Package | IMPORT | com.vmware.pscoe.library.ecmascript-2.38.2-SNAPSHOT.package
[INFO] Package | IMPORT | com.test.test-1.0.4-SNAPSHOT.package
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
```

### Release Notes

This strategy will force you to upload the same or newer version of a package, otherwise it will fail the build, allowing us for 
better CI/CD pipelines, where we can ensure that the latest versions are always used on the server.

The new strategy can be triggered by passing `-Dvro.forceImportLatestVersions=true`. It is by default set to `true`.

Example usage:
```bash
mvn clean package vrealize:push -DincludeDependencies=true -Dvro.forceImportLatestVersions=true -DskipTests -PDevLab
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+### Enhancements
+
+* [vro] Added new strategy StrategyForceLatestVersions that will force you to upload a newer or same version of artifacts, otherwise the build will fail. New property has been introduced `-Dvro.forceImportLatestVersions={{BOOLEAN}}` that is set by default to true
```
Comment (VenelinBakalov) [inline CHANGELOG.md:3]: In the release.md it says the strategy is by default to false: "It is by default set to `false`." I think it looks as if false is the actual value but could you double check and update the changelog?

Comment (Michaelpalacce) [reply CHANGELOG.md:3]: Oh... good catch 😆 should be by default to false

Comment (VenelinBakalov) [reply CHANGELOG.md:3]: Nice, that would be great as it is also backwards compatible

Comment (Michaelpalacce) [reply CHANGELOG.md]: Yeah, that was the main philosophy 😮‍💨 even tho I think some improvements to the way strategies work are needed if we are to add more

File: .github/linters/sun_checks.xml [hunk 2]
```diff
@@ -205,7 +205,7 @@ https://www.oracle.com/java/technologies/javase/codeconventions-contents.html
 		<module name="ArrayTypeStyle" />
 		<!-- FinalParameters reports False positives -->
 		<!-- <module name="FinalParameters" /> -->
-		<module name="TodoComment" />
+		<!-- <module name="TodoComment" /> -->
```
Comment (VenelinBakalov) [inline .github/linters/sun_checks.xml:208]: Do we need this comment?

**Review by VenelinBakalov — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (VenelinBakalov) [reply CHANGELOG.md]: Hmmm if you have specific idea for that maybe we can try to implement and release together with Anton's typescript update as major version

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+### Enhancements
+
+* [vro] Added new strategy StrategyForceLatestVersions that will force you to upload a newer or same version of artifacts, otherwise the build will fail. New property has been introduced `-Dvro.forceImportLatestVersions={{BOOLEAN}}` that is set by default to false
```
Comment (joroaf) [inline CHANGELOG.md:3]: As we spoke on the roadmap meeting - let's make it default == true

**Review by VenelinBakalov — APPROVED** [review]

**Comment by pe1pip** [comment]
would it be possible to just exclude the `com.vmware.pscoe.library.ecmascript` from this policy? We have a few environments with newer versions of this lib that we are currently using, for example due to the use of  newer BT versions for some packages that we decided against for other packages.

**Review by joroaf — APPROVED** [review]
As we agreed, we can disable this strategy's default behavior. We are going to introduce new version 3 where this strategy will be enabled by default.

**Pull request merged**